pkg/localcontroller/managers/incrementallearning/incrementallearningjob.go

```diff
@@ -700,37 +749,48 @@ func (im *Manager) getEvalResult(job *Job) ([]map[string][]float64, error) {
 	return result, err
 }
 
-// getJobStageModel gets model from job conditions for train/eval/deploy
-func (im *Manager) getJobStageModel(job *Job, jobStage sednav1.ILJobStage) []Model {
+// getModelFromJobConditions gets model from job conditions for train/eval/deploy
+func (im *Manager) getModelFromJobConditions(job *Job, jobStage sednav1.ILJobStage) *Model {
 	jobConditions := job.Status.Conditions
+	jobConfig := job.JobConfig
 
+	getModel := func(initModel *Model, models []Model) *Model {
+		for _, m := range models {
+			if m.Format == initModel.Format {
+				if initModel.Devices != nil && len(m.Devices) == 1 {
+					for _, d := range initModel.Devices {
+						if m.Devices[0] == d {
+							return &m
+						}
+					}
+				} else {
+					return &m
+				}
+			}
+		}
+
+		return nil
+	}
+
+	models := im.getModelsFromJobConditions(jobConditions, sednav1.ILJobTrain, sednav1.ILJobStageCondCompleted, "output")
+	if models == nil {
+		return nil
+	}
+
+	var model *Model
 	switch jobStage {
 	case sednav1.ILJobTrain:
-		// the second model is the pre-trained model of train stage.
-		models := im.getModelsFromJobConditions(jobConditions, sednav1.ILJobTrain, sednav1.ILJobStageCondCompleted, "output")
-		if models != nil {
-			return []Model{{Format: models[1].Format, URL: models[1].URL}}
-		}
+		model = jobConfig.TrainModel
 	case sednav1.ILJobEval:
-		// the first model is the output model of train stage.
-		models := im.getModelsFromJobConditions(jobConditions, sednav1.ILJobTrain, sednav1.ILJobStageCondCompleted, "output")
-		if models != nil {
-			return []Model{{Format: models[0].Format, URL: models[0].URL}}
-		}
+		model = jobConfig.EvalModel
 	case sednav1.ILJobDeploy:
-		// two models for deploy stage:
-		// the first model is the output model of train stage, which was evaluated as better than the second model in eval stage.
-		// the second model is the serving model used in the inference worker.
-		var deployModels []Model
-		models := im.getModelsFromJobConditions(jobConditions, sednav1.ILJobEval, sednav1.ILJobStageCondReady, "input")
-		for _, m := range models {
-			deployModels = append(deployModels, Model{Format: m.Format, URL: m.URL})
-		}
-
-		return deployModels
+		model = jobConfig.DeployModel
+	}
+	if model == nil {
```

Comment:
It is redundant to write another judgment statement, put it in the default syntax block of switch：
```
switch jobStage {
    case xxx:
    ......
    default:
        return nil
}
```

Reply:
jobStage value can not be `default`;

Reply:
got it